src/freertos_drivers/common/WiFiInterface.hxx

```diff
@@ -0,0 +1,250 @@
+    /// @param dns_ip is the DNS server IP address for the access point to
+    /// provide to stations.
+    void setup_access_point(const std::string &ssid,
+                            const std::string &security_key = "",
+                            WiFiSecurity security_type = WiFiSecurity::OPEN,
+                            uint32_t ap_ip = 0, uint32_t dns_ip = 0);
+
+    /// Initiate the WPS Push Button Control connection process.
+    ///
+    /// @param timeout is the number of milliseconds to wait for the WPS
+    /// process to complete. A value of zero indicates that the interface
+    /// default timeout should be used.
+    void initiate_wps_connect(const uint32_t timeout = 0);
+
+    /// @return the current WiFi interface operating mode.
+    WiFiMode get_mode();
+
+    /// @return the current WiFi interface status.
+    WiFiState get_state();
+
+    /// Adds a saved SSID profile.
+    ///
+    /// @param ssid SSID of the profile to save.
+    /// @param sec_type @ref WiFiSecurity of the profile to be saved.
+    /// @param key password of the SSID, nullptr allowed if sec_type is OPEN.
+    /// @param priority connection priority when more than one of the saved
+    /// networks is available, 0 == lowest priority
+    /// @return resulting index in the list of profiles, else -1 on error
+    int add_ssid_profile(const std::string &ssid, const WiFiSecurity sec_type,
+                         const std::string &key, const uint8_t priority);
+
+    /// Delete a saved SSID profile by it's SSID.
+    ///
+    /// @param ssid SSID of the profile to delete.
+    /// @return 0 upon success, else -1 on error.
+    int del_ssid_profile(const std::string &ssid);
+
+    /// Delete a saved SSID profile.
+    ///
+    /// @param index index within saved profile list to remove, a value of 0xFF
+    /// will remove all stored profiles.
+    /// @return 0 upon success, else -1 on error.
+    int del_ssid_profile(const int index);
+
+    /// Retrieves a saved SSID profile by index.
+    ///
+    /// @param index of the stored profile to retrieve.
+    /// @param ssid is a 33 byte array containing the ssid of the profile.
+    /// @param sec_type is the security type of the profile.
+    /// @param priority is the priority of the profile.
+    /// @return 0 upon success, else -1 on error.
+    int get_ssid_profile(const int index, const std::string &ssid,
+                         const WiFiSecurity &sec_type,
+                         const uint8_t &priority);
+
+    /// @return true if there are no saved SSID profiles, false otherwise.
+    bool are_ssid_profiles_empty();
+
+    /// Retrieves a list of available access points.
+    ///
+    /// @param entries will contain a list of available network entries.
+    /// @param count is the number of network entries to retrieve, max of 20.
+    /// @return number of network entries retrieved.
+    int get_network_list(NetworkListEntry *entries, size_t count);
```

Comment:
I was thinking the same but modelled after CC32xx API. I'm almost thinking just return the vector instead as it can be used for iteration in that case

Reply:
Adjusting to std::vector<NetworkEntryList> *